api/src/main/java/io/opentelemetry/trace/propagation/HttpTraceContext.java

```diff
@@ -93,9 +93,19 @@
     chars[0] = VERSION.charAt(0);
     chars[1] = VERSION.charAt(1);
     chars[2] = TRACEPARENT_DELIMITER;
-    spanContext.getTraceId().copyLowerBase16To(chars, TRACE_ID_OFFSET);
+
+    CharSequence traceId = spanContext.getTraceId();
+    for (int i = 0; i < traceId.length(); i++) {
```

Comment:
Any specific reason to use `CharSequence`? Looks like we lose being able to use `getChars` and not sure we gain any benefit in return.